src/analyses/controllers.clj

```diff
@@ -6,16 +6,19 @@
 
 (defn add-quicklaunch
   [user quicklaunch]
-  (clients/validate-submission
-   {:quicklaunch quicklaunch
-    :app         (clients/get-app user system-id (:app_id quicklaunch))
-    :system-id   system-id
-    :user        user})
-  (persist/add-quicklaunch user quicklaunch))
+  (let [{:keys [version_id] :as app} (clients/get-app user system-id (:app_id quicklaunch))
+        quicklaunch                  (assoc quicklaunch :app_version_id version_id)]
+    (clients/validate-submission
+      {:quicklaunch quicklaunch
+       :app         app
+       :system-id   system-id
+       :user        user})
+    (persist/add-quicklaunch user quicklaunch)))
 
 (defn update-quicklaunch
   [id user quicklaunch]
   (let [ql-merged  (merge (persist/get-quicklaunch id user) quicklaunch)
+        ;FIXME     Lookup by app version ID
```

Comment:
If I'd gotten this far in the review yesterday, I would have realized that more stuff was coming later. 😆 🤦‍♀️